yt_dlp/extractor/youtube.py

```diff
@@ -1462,6 +1489,226 @@ def _extract_yt_initial_variable(self, webpage, regex, video_id, name):
+
+                    # Youtube sometimes sends incomplete data
+                    # See: https://github.com/ytdl-org/youtube-dl/issues/28194
+                    last_error = 'Incomplete data received'
+                    if count >= retries:
+                        self._downloader.report_error(last_error)
+
+            if not response:
+                break
+
+            known_continuation_renderers = {
+                'itemSectionContinuation': extract_thread,
+                'commentRepliesContinuation': extract_thread
+            }
+
+            # extract next root continuation from the results
+            continuation_contents = try_get(
+                response, lambda x: x['continuationContents'], dict) or {}
+
+            for key, value in continuation_contents.items():
+                if key not in known_continuation_renderers:
+                    continue
+                continuation_renderer = value
+
+                if first_continuation:
+                    expected_video_comment_count = try_get(continuation_renderer,
+                                                           (lambda x: x['header']['commentsHeaderRenderer']['countText']['runs'][0]['text'],
+                                                            lambda x: x['header']['commentsHeaderRenderer']['commentsCount']['runs'][0]['text']),
+                                                           compat_str)
+                    if expected_video_comment_count:
+                        # TODO: is this still needed?
+                        expected_video_comment_count = expected_video_comment_count.replace(' Comments', '').replace('1 Comment', '1').replace(',', '')
+                        expected_video_comment_count = str_to_int(expected_video_comment_count)
+                        self.to_screen("Downloading ~%d comments" % expected_video_comment_count)
+                        first_continuation = False
+
+                for entry in known_continuation_renderers[key](continuation_renderer):
+                    yield entry
+
+                continuation = YoutubeTabIE._extract_continuation(continuation_renderer)  # TODO
+                break
+
+    def _extract_comments(self, ytcfg, video_id, contents, webpage, xsrf_token):
+        """Entry for comment extraction"""
+        comments = []
+        known_entry_comment_renderers = (
+            'itemSectionRenderer',
+        )
+        for entry in contents:
+            for key, renderer in entry.items():
+                if key not in known_entry_comment_renderers:
+                    continue
+
+                comment_iter = self._comment_entries(renderer,
+                                                     identity_token=self._extract_identity_token(webpage, item_id=video_id),
+                                                     account_syncid=self._extract_account_syncid(ytcfg),
+                                                     session_token_list=[xsrf_token])
+
```

Comment:
I prefer this style of code. (No need to change, just a matter of personal preference)
```py
comment_iter = self._comment_entries(
    renderer,
    identity_token=self._extract_identity_token(webpage, item_id=video_id),
    ...)
```